### PR TITLE
Fix: generator methods in classes (fixes #85)

### LINKED ISSUE #85: Parser fails on generator methods
Espree parser fails on the following with `Unexpected token *`:

```javascript
class Foo {
  *bar() {
  }
}
```

Test case:

```javascript
var espree = require("espree");

espree.parse('class Foo {*bar() {}}', {
  "ecmaFeatures": {
    "classes": true,
    "generators": true,
    "objectLiteralShorthandMethods": true,
  },
});
```

Stack:

```
node_modules/espree/espree.js:5373
        throw e;
              ^
Error: Line 1: Unexpected token *
    at throwError (node_modules/espree/espree.js:2078:17)
    at throwUnexpected (node_modules/espree/espree.js:2141:5)
    at parseObjectPropertyKey (node_modules/espree/espree.js:2364:5)
    at parseClassBody (node_modules/espree/espree.js:4934:15)
    at parseClassDeclaration (node_modules/espree/espree.js:5023:17)
    at parseSourceElement (node_modules/espree/espree.js:5053:28)
    at parseSourceElements (node_modules/espree/espree.js:5101:25)
    at parseProgram (node_modules/espree/espree.js:5120:12)
    at Object.parse (node_modules/espree/espree.js:5361:19)
    at Object.<anonymous> (esp.js:3:8)
```

See eslint/eslint#1990.

### PR DESCRIPTION
**Pull request by xjamundx**
- [x] supports generator methods in classes
- [x] supports static generator methods in classes
- [x] tests the success cases
- [x] tests to ensure failure when `generators` are `false`
- [x] tests to ensure failure when `generators` are `true` and `objectLiteralShorthandMethods` is `false`

### CODE REVIEW

**Comment by nzakas** [comment]
Looks good. Can you add a test to be sure an error occurs when `ecmaFeatures.classes` is false and `ecmaFeatures.objectLiteralShorthandMethods` is false?

**Comment by xjamundx** [comment]
Yes, will get those in there ASAP

**Comment by xjamundx** [comment]
Pretty sure you just found a bug :)

**Comment by xjamundx** [comment]
What about ES6 `static` class methods that are also generators?

File: espree.js [hunk 1]
```diff
@@ -4929,10 +4943,25 @@ function parseClassBody() {
...
+            if (!allowGenerators) {
+                throwUnexpected(lookahead);
+            }
+            lex();
+        }
+
         key = parseObjectPropertyKey();
 
+        // static generator methods
+        if (key.name === "static" && match("*")) {
```
Comment (xjamundx) [inline espree.js]: need a test for the negative case here as well....1 sec

Comment (xjamundx) [inline espree.js]: done...

**Comment by xjamundx** [comment]
Okay I *think* this is fairly good now. Any feedback would be appreciated :)

**Comment by nzakas** [comment]
LGTM! Ready to merge?

**Comment by xjamundx** [comment]
:)
